packages/nimble-components/src/spinner/specs/README.md

```diff
@@ -0,0 +1,125 @@
+# Nimble Spinner
+
+## Overview
+
+Spinner component (`nimble-spinner`): Indeterminate progress indicator / loading indicator. Meant to be temporarily used in a region of a web page until that region's contents have loaded, or to represent an ongoing operation that doesn't have a known/ determinate duration.
+
+### Background
+
+[Nimble spinner: #346](https://github.com/ni/nimble/issues/346)  
+[Nimble spinner design: #822](https://github.com/ni/nimble/issues/822)  
+[Visual Design spec - Adobe XD](https://xd.adobe.com/view/33ffad4a-eb2c-4241-b8c5-ebfff1faf6f6-66ac/screen/dece308f-79e7-48ec-ab41-011f3376b49b/)
+
+### Non-goals
+
+Ability to show determinate progress:
+- The current visual design for the spinner is distinct/different from an indeterminate radial progress bar
+- We don't currently have visual designs for a linear or radial progress bar with determinate progress
+- If/when we want progress bar components in the future, at that point we'll probably create new Nimble components which extend the corresponding FAST components (`fast-progress` for a linear progress bar, and `fast-progress-ring` for a radial progress bar)
+  
+### Features
+
+- Show indeterminate progress with an animation
+  - Cannot be paused - assumption is that component will be added to/ removed from the page as needed
+- Supports 3 sizes: `small` (16x16), `medium` (32x32) (default), `large` (64x64)
+- For the color theme only, supports an alternate appearance/color scheme via `appearance-variant="alternate"`
+
+### Risks and Challenges
+
+None
+
+### Prior Art/Examples
+
+(TODO) *Screenshots and/or links to existing, canonical, or exemplary implementations of the component.*
+
+## Design
+
+The `nimble-spinner` is an animating indicator that can be placed in a particular region of a page to represent loading progress, or an ongoing operation, of an indeterminate / unknown duration.
+
+### API
+
+Component Name: `nimble-spinner`
```

Comment:
If we think we may someday support a radial spinner, should we call this `nimble-bit-spinner` or something like that? 

I guess it depends whether:
1. this will be the "official" spinner and the radial spinner be a variant for a few specific use cases. In this case I'd be ok with `nimble-spinner` for this one and `nimble-radial-spinner` for the other one
2. both will be commonly used, just for different purposes. In this case I'd vote for `nimble-bit-spinner` and `nimble-radial-spinner` to reflect the fact that they're "peers"

My guess is that, if we bother to implement a radial spinner, it will be commonly used so 2 is more likely. But @nate-ni might have the best guess about possible future directions (or be able to inform based on what other design systems do).

Reply:
I guess I was thinking that we'd match FAST's naming (i.e. `nimble-progress` and `nimble-progress-ring`), but `nimble-linear-progress` and `nimble-radial-progress` would also work.

I think that the linear + radial progress components would be the 'peers' (same API, support determinate + indeterminate, just different visual representations), and this spinner is the 1-off basic one (so it doesn't need `progress` in the name).

Reply:
Added my future naming expectations to the spec, and added a bullet point on finalizing the naming in Open Issues for now.

Reply:
I don't have a strong enough opinion to veto your proposal of calling this a spinner and the others variations on "progress". 

However, to me they're all used for similar purposes so it would make sense for them to have similar names (acknowledging that only some of them will support determinate progress).

A couple other options would be something like:
1. `nimble-bits-progress`, `nimble-linear-progress`, and `nimble-radial-progress`
2. `nimble-progress-bits`, `nimble-progress-bar`, and `nimble-progress-ring`

Reply:
This is great input. 

I think this particular control should be `nimble-spinner`

The others - 
`nimble-progress-ring` (Determinate and indeterminate radial progress)
`nimble-progress-bar` (Determinate and indeterminate bar progress)

Reply:
> Do we think the different flavors of indeterminate or determinate progress controls will have significantly different layout or property behaviors? In addition do we think there will be "many" or that the size of them will be significant?

I think we pretty much already know all the variations: Appearance (linear progress bar, radial progress bar, 'other' like this "bits" spinner) x (determinate progress, indeterminate progress)

There's lots of variations in the 'other' category though ([https://loading.io/css/](https://loading.io/css/) shows some, most only make sense for indeterminate progress)

I want to point out that it's simple enough to support determinate + indeterminate in the same control. [FAST does that](https://github.com/microsoft/fast/tree/master/packages/web-components/fast-foundation/src/progress) - if you don't specify `value` it's indeterminate, if you do it's determinate. API wise linear+radial appearances could easily be in the same component (again looking at FAST, their implementations are identical / all in a base class).

I'm not sure that it really feels like we're reaching a consensus yet. I don't think we need to settle the future progress component names now, we just need to confirm if we fine with `nimble-spinner` for this one (feels like 'mostly yes' right now)

Reply:
Agree with Malcolm that it's good to separate this component from the "progress" counterparts semantically, in that this component has no determinate mode. Using the "progress" vs "spinner" terms seems sufficient in this sense.

Reply:
@msmithNI, I think you are right about us mostly being good to go with `<nimble-spinner>`. 

Go forth and conquer 😁

Reply:
Removed component name from Open Issues (going with `nimble-spinner`)